src/core/segment_reader.rs

```diff
@@ -69,6 +70,17 @@ impl SegmentReader {
         &self.schema
     }
 
+    /// Merges the passed bitset with the existing one.
+    pub fn apply_delete_bitset(&mut self, delete_bitset: DeleteBitSet) {
```

Comment:
Let's not mutate `SegmentReader`

Reply:
I would prefer this to the alternative of having (segment_reader, alive_bitset) tuples in the merger everywhere